cmd/kubeadm/app/apis/kubeadm/validation/validation.go

```diff
@@ -291,8 +291,11 @@ func ValidateEtcd(e *kubeadm.Etcd, fldPath *field.Path) field.ErrorList {
 func ValidateCertSANs(altnames []string, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
 	for _, altname := range altnames {
-		if len(validation.IsDNS1123Subdomain(altname)) != 0 && net.ParseIP(altname) == nil {
+		if errs := validation.IsDNS1123Subdomain(altname); len(errs) != 0 && net.ParseIP(altname) == nil {
 			allErrs = append(allErrs, field.Invalid(fldPath, altname, "altname is not a valid dns label or ip address"))
+			for _, err := range errs {
+				allErrs = append(allErrs, field.Invalid(fldPath, altname, err))
+			}
```

Comment:
@SataQiu shouldn't this be broken down like so?

```
if errs := validation.IsDNS1123Subdomain(altname); len(errs) != 0 {
	for _, err := range errs {
		allErrs = append(allErrs, field.Invalid(fldPath, altname, err))
	}
}
if net.ParseIP(altname) == nil {
	allErrs = append(allErrs, field.Invalid(fldPath, altname, "altname is not a valid dns label or ip address"))
}
```

Reply:
@neolit123 thank you, I have changed it.

Reply:
we can also uppercase dns and ip (sorry, forgot about this).
`altname is not a valid DNS label or IP address`
but just a nit..